samples/subsys/mgmt/updatehub/src/main.c

```diff
@@ -95,15 +95,15 @@ void main(void)
 			       server_certificate,
 			       sizeof(server_certificate)) < 0) {
 		LOG_ERR("Failed to register server certificate");
-		return;
+		return 0;
 	}
 
 	if (tls_credential_add(CA_CERTIFICATE_TAG,
 			       TLS_CREDENTIAL_PRIVATE_KEY,
 			       private_key,
 			       sizeof(private_key)) < 0) {
 		LOG_ERR("Failed to register private key");
-		return;
+		return 0;
```

Comment:
Hi @keith-packard ,

I noted that these two init errors not return a value. Should those return a -errno code ?

Reply:
Zephyr explicitly ignores the return value from main(). Maybe the documentation patch should make it explicit that a return value of zero is expected?

Reply:
Since Zephyr don't care about the return value you can ignore my previous comment. Documentation may be good to avoid same questions on the communication channels.

Reply:
good plan. I've added a sentence to both the c and c++ documents indicating that Zephyr ignores the return value from main and that applications should return zero.